packages/flutter/lib/src/rendering/mouse_cursor.dart

```diff
@@ -0,0 +1,419 @@
+  const SystemMouseCursor._({
+    @required this.kind,
+    @required this.debugDescription,
+  }) : assert(kind != null),
+       assert(debugDescription != null);
+
+  /// A string that identifies the kind of the cursor.
+  ///
+  /// The interpretation of [kind] is platform-dependent.
+  final String kind;
+
+  @override
+  final String debugDescription;
+
+  @override
+  @protected
+  _SystemMouseCursorSession createSession(int device) => _SystemMouseCursorSession(this, device);
+
+  @override
+  bool operator ==(dynamic other) {
+    if (other.runtimeType != runtimeType)
+      return false;
+    return other is SystemMouseCursor
+        && other.kind == kind;
+  }
+
+  @override
+  int get hashCode => kind.hashCode;
+
+  @override
+  void debugFillProperties(DiagnosticPropertiesBuilder properties) {
+    super.debugFillProperties(properties);
+    properties.add(DiagnosticsProperty<String>('kind', kind, level: DiagnosticLevel.debug));
+  }
+}
+
+/// A collection of system [MouseCursor]s.
+///
+/// System cursors are standard mouse cursors that are provided by the current
+/// platform. They don't require external resources.
+///
+/// [SystemMouseCursors] is a superset of the system cursors of every platform
+/// that Flutter supports, therefore some of these objects might map to the same
+/// result, or fallback to the basic arrow. This mapping is defined by the
+/// Flutter engine.
+///
+/// The cursor names are chosen to reflect the cursors' use cases instead of
+/// their shapes, because different platforms might (although not commonly) use
+/// different shapes for the same use case.
+class SystemMouseCursors {
+  // This class only contains static members, and should not be instantiated or
+  // extended.
+  factory SystemMouseCursors._() => null;
+
+  /// A special value that doesn't change cursor by itself, but make a region
+  /// that blocks other regions behind it from changing the cursor.
+  ///
+  /// When a pointer enters a region with a cursor of [uncontrolled], the pointer
+  /// retains its previous cursor and keeps so until it moves out of the region.
+  /// Technically, this region absorb the mouse cursor hit test without changing
+  /// the pointer's cursor.
+  ///
+  /// This is useful in a region that displays a platform view, which let the
+  /// operating system handle pointer events and change cursors accordingly. To
+  /// achieve this, the region's cursor must not be any Flutter cursor, since
+  /// that might overwrite the system request upon pointer entering; the cursor
+  /// must not be null either, since that allows the widgets behind the region to
+  /// change cursors.
+  static const MouseCursor uncontrolled = _NoopMouseCursor._();
+
+  /// Hide the cursor.
+  ///
+  /// Any cursor other than [none] or [uncontrolled] unhides the cursor.
+  static const SystemMouseCursor none = SystemMouseCursor._(kind: 'none', debugDescription: 'none');
```

Comment:
Maybe the debugDescription should just default to the kind string for SystemMouseCursor to avoid the duplication here?